# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PermissionMessageUtils.cpp
# Commit: af27682c5e7f
# Full Hash: af27682c5e7f86ff17d6ed152c8010e763870bdc
# Author: ssengupta <ssengupta@mozilla.com>
# Date: 2020-05-13 03:44:05
# Regressor Bug: 1635399
# File Overlap Count: 1
# Description:
#   Bug 1635399 - Function PrincipalInfoToPrincipal now returns Result<nsCOMPtr<nsIPrincipal>, nsresult> r=ckerschb
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73868
# ==============================================================================

diff -r 86b22b1a9f7e -r af27682c5e7f dom/ipc/PermissionMessageUtils.cpp
--- a/dom/ipc/PermissionMessageUtils.cpp	Tue May 12 23:55:42 2020 +0300
+++ b/dom/ipc/PermissionMessageUtils.cpp	Tue May 12 19:02:05 2020 +0000
@@ -7,6 +7,8 @@
 #include "mozilla/dom/PermissionMessageUtils.h"
 #include "mozilla/ipc/BackgroundUtils.h"
 #include "mozilla/ipc/PBackgroundSharedTypes.h"
+#include "nsCOMPtr.h"
+#include "nsIPrincipal.h"
 
 namespace mozilla {
 namespace ipc {
@@ -35,9 +37,19 @@
     return false;
   }
 
-  nsresult rv = NS_OK;
-  *aResult = info ? PrincipalInfoToPrincipal(info.ref(), &rv) : nullptr;
-  return NS_SUCCEEDED(rv);
+  if (info.isNothing()) {
+    return true;
+  }
+
+  auto principalOrErr = PrincipalInfoToPrincipal(info.ref());
+
+  if (NS_WARN_IF(principalOrErr.isErr())) {
+    return false;
+  }
+
+  nsCOMPtr<nsIPrincipal> principal = principalOrErr.unwrap();
+  *aResult = principal;
+  return true;
 }
 
 }  // namespace ipc